python/ray/data/datasource/databricks_uc_datasource.py

```diff
@@ -0,0 +1,187 @@
+                urljoin(url_base, f"{statement_id}/cancel"),
+                auth=req_auth,
+                headers=req_headers,
+            )
+            response.raise_for_status()
+            raise
+
+        if state != "SUCCEEDED":
+            raise RuntimeError(f"Query {self.query} execution failed.")
+
+        manifest = response.json()["manifest"]
+        is_truncated = manifest['truncated']
+
+        if is_truncated:
+            logger.warning(
+                f"The result dataset of '{query}' exceeding 100GiB and it is truncated."
+            )
+
+        chunks = manifest["chunks"]
+
+        # Make chunks metadata are ordered by index.
+        chunks = sorted(chunks, key=lambda x: x['chunk_index'])
+        num_chunks = len(chunks)
+        self.num_chunks = num_chunks
+        self._estimate_inmemory_data_size = sum(
+            chunk['byte_count'] for chunk in chunks
+        )
+
+        def get_read_task(task_index, parallelism):
+            # 0 <= task_index < parallelism
+            chunk_index_list = list(range(task_index, parallelism, num_chunks))
+
+            num_rows = sum(
+                chunks[chunk_index]['row_count'] for chunk_index in chunk_index_list
+            )
+            size_bytes = sum(
+                chunks[chunk_index]['byte_count'] for chunk_index in chunk_index_list
+            )
+
+            metadata = BlockMetadata(
+                num_rows=num_rows,
+                size_bytes=size_bytes,
+                schema=None,
+                input_files=None,
+                exec_stats=None,
+            )
+
+            def read_fn():
+                for chunk_index in chunk_index_list:
+                    chunk_info = chunks[chunk_index]
+                    row_offset_param = urlencode({'row_offset': chunk_info["row_offset"]})
+                    resolve_external_link_url = urljoin(
+                        url_base,
+                        f"{statement_id}/result/chunks/{chunk_index}?{row_offset_param}"
+                    )
+
+                    resolve_response = requests.get(
+                        resolve_external_link_url,
+                        auth=req_auth,
+                        headers=req_headers
+                    )
+                    resolve_response.raise_for_status()
+                    external_url = resolve_response.json()["external_links"][0]["external_link"]
+                    # NOTE: do _NOT_ send the authorization header to external urls
+                    raw_response = requests.get(external_url, auth=None, headers=None)
+                    raw_response.raise_for_status()
+
+                    arrow_table = pyarrow.ipc.open_stream(raw_response.content).read_all()
+                    yield arrow_table
+
+            return ReadTask(read_fn=read_fn, metadata=metadata)
+
+        self._get_read_task = get_read_task
+
+    def estimate_inmemory_data_size(self) -> Optional[int]:
+        return self._estimate_inmemory_data_size
```

Comment:
```suggestion
    def estimate_inmemory_data_size(self) -> int:
        return self._estimate_inmemory_data_size
```

looks like `self._estimate_inmemory_data_size` always int.

Reply:
just to keep the same API signature with base class

Reply:
makes sense